generate_buildjl.jl

```diff
@@ -87,10 +87,42 @@ function _repr(p::Product)
     return replace(repr(p), "(" => "(prefix, ")
 end
 
+function compat_string_of_platform_key_abi(platform)
+     if platform == "aarch64-linux-gnu"
+         return "Linux(:aarch64, libc=:glibc)"
+     elseif  platform == "aarch64-linux-musl"
+         return "Linux(:aarch64, libc=:musl)"
+     elseif platform == "armv7l-linux-gnueabihf"
+         return "Linux(:armv7l, libc=:glibc, call_abi=:eabihf)"
+     elseif platform == "armv7l-linux-musleabihf"
+         return "Linux(:armv7l, libc=:musl, call_abi=:eabihf)"
+     elseif platform == "i686-linux-gnu"
+         return "Linux(:i686, libc=:glibc)"
+     elseif platform == "i686-linux-musl"
+         return "Linux(:i686, libc=:musl)"
+     elseif platform == "powerpc64le-linux-gnu"
+         return "Linux(:powerpc64le, libc=:glibc)"
+     elseif platform == "x86_64-apple-darwin"
+         return "MacOS(:x86_64)"
+     elseif  platform == "x86_64-linux-gnu"
+         return "Linux(:x86_64, libc=:glibc)"
+     elseif platform == "x86_64-linux-musl"
+         return "Linux(:x86_64, libc=:musl)"
+     elseif platform == "x86_64-unknown-freebsd"
+         return "FreeBSD(:x86_64)"
+     elseif platform == "x86_64-w64-mingw32"
+         return "Windows(:x86_64)"
+     elseif platform == "i686-w64-mingw32"
+         return "Windows(:i686)"
+     else
+         error("Platform $platform not known")
+     end
+end
+
 # `Pkg.CompilerABI` has diverged from `BinaryProvider.CompilerABI`.
 # This function fixes common differences.
 function _pkey(platform)
-    pkey = string(platform_key_abi(platform))
+    pkey = string(compat_string_of_platform_key_abi(platform))
```

Comment:
Isn't this
```suggestion
    pkey = string(parse(Platform, platform))
```
?

Reply:
The platforms are nowadays printed differently and the output cannot be parsed by julia. Why is your version different to what was there before?

Reply:
Then you want
```suggestion
    pkey = repr(parse(Platform, platform))
```

Reply:
Oh, I see, this is to get the _old_ representation

Reply:
Indeed. I took the lazy approach.